### ISSUE TITLE
typing.Text not available in python 3.5.1

### ISSUE THREAD

**Issue by 1225713f-2075-4ece-83e8-9cfea2da1b63** [python/cpython#74065]
BPO | [29879](https://bugs.python.org/issue29879)
--- | :---
Nosy | @DimitrisJim, @Mariatta, @cblegare
PRs | <li>python/cpython#784</li><li>python/cpython#4573</li><li>python/cpython#4574</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-11-26.22:30:30.806>
created_at = <Date 2017-03-22.14:11:25.213>
labels = ['3.7', 'docs']
title = 'typing.Text not available in python 3.5.1'
updated_at = <Date 2017-11-26.22:30:30.803>
user = 'https://github.com/cblegare'
```

bugs.python.org fields:
```python
activity = <Date 2017-11-26.22:30:30.803>
actor = 'Mariatta'
assignee = 'docs@python'
closed = True
closed_date = <Date 2017-11-26.22:30:30.806>
closer = 'Mariatta'
components = ['Documentation']
creation = <Date 2017-03-22.14:11:25.213>
creator = 'Charles Bouchard-L\xc3\xa9gar\xc3\xa9'
dependencies = []
files = []
hgrepos = []
issue_num = 29879
keywords = ['patch']
message_count = 9.0
messages = ['289985', '289986', '289987', '289994', '289995', '289997', '307022', '307023', '307024']
nosy_count = 4.0
nosy_names = ['docs@python', 'Jim Fasarakis-Hilliard', 'Mariatta', 'Charles Bouchard-L\xc3\xa9gar\xc3\xa9']
pr_nums = ['784', '4573', '4574']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue29879'
versions = ['Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by 1225713f-2075-4ece-83e8-9cfea2da1b63** [comment]
The real issue here is that this is not documented in [Doc/library/typing.rst](https://github.com/python/cpython/blob/main/Doc/library/typing.rst).

**Comment by b78688f9-6435-4192-9fe9-2f86f7e61fcf** [comment]
I'm guessing you might of missed it, `Text` is documented in the docs for Python 3.5 https://docs.python.org/3.5/library/typing.html#typing.Text :-)

**Comment by 1225713f-2075-4ece-83e8-9cfea2da1b63** [comment]
I mean, what is not documented is 'New in 3.5.2'

**Comment by b78688f9-6435-4192-9fe9-2f86f7e61fcf** [comment]
Ah I see now, yes, neither are Reversible, Type, NewType, TYPE_CHECKING and DefaultDict. 

Why don't you go ahead an submit a PR for this? :-)

**Comment by 1225713f-2075-4ece-83e8-9cfea2da1b63** [comment]
Thank you for confirming this.  I'll look into it

**Comment by b78688f9-6435-4192-9fe9-2f86f7e61fcf** [comment]
That's great, make sure you also take a look at the quick start section of the [devguide](https://devguide.python.org) [1] if you need help in any steps :-)

[1]: https://docs.python.org/devguide/#quick-start

**Comment by Mariatta** [comment]
New changeset 0cd2e81bea639828d7c9a7afc61fb1da9699492c by Mariatta (Ivan Levkivskyi) in branch 'master':
bpo-29879: Update typing documentation. (GH-4573)
https://github.com/python/cpython/commit/0cd2e81bea639828d7c9a7afc61fb1da9699492c

**Comment by Mariatta** [comment]
New changeset cceb0f8d7b3451fb8de03651c979f5ac639b68c0 by Mariatta (Miss Islington (bot)) in branch '3.6':
bpo-29879: Update typing documentation. (GH-4573) (GH-4574)
https://github.com/python/cpython/commit/cceb0f8d7b3451fb8de03651c979f5ac639b68c0

**Comment by Mariatta** [comment]
This has been fixed in the 3.7 and 3.6 branches.
Thanks all!
